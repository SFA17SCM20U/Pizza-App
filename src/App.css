
.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 30vmin;
  pointer-events: none;
  display: block;
  margin: auto;
}

.App-header {
  background-color: #dee35d;
  height: 300px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(20px + 5vmin);
  font-family: Comic Sans MS;
  color: white;
  box-shadow: 0 0 0.8125rem 0 rgba(0,0,0,.75);
}

.img-size{
  width: 450px !important;
  height: 350px !important;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.bg-img-class{
  position: absolute;
  z-index: -1;
  opacity: 0.4;
  width: 100%;
}

.center {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.footer{
  background-color: #dee35d;
  height:50px;
  text-align: center;
  margin: auto;
}

.footer-attr{
  margin-top: 60px;
  left: 0px;
  bottom: 0px;
  width: 100%;
}
