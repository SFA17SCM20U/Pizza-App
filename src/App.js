import React, { Component } from 'react';
import './App.css';
import PizzaComponent from './components/pizzacomponents.js';
import Logo from './images/logo.png';

class App extends Component {
  render() {
    return (
      <div>
        <div className="App-header">
          <img src={Logo} alt="PizzaTym"/>          
        </div>
          <PizzaComponent>

          </PizzaComponent>
          <div className="footer-attr">
            <footer className="footer">
              <span><b>Copyright Â©2019 PizzaTym, USA. All rights reserved.</b></span>
            </footer>
          </div>
      </div>
    );
  }
}

export default App;
