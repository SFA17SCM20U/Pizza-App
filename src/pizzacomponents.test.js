import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import renderer from 'react-test-renderer';
import 'jest-dom/extend-expect';
import PizzaComponent from './components/pizzacomponents.js';
import Pizza from './components/pizza.js';
import App from "./App.js";
import { resolve } from 'dns';
import { reject } from 'q';

afterEach(cleanup);

//UI component test
it('renders loading message', () => {
  const { getByText } = render(<PizzaComponent />);
  expect(getByText('Loading data...')).toBeInTheDocument();
  expect(getByText('Loading data...')).toMatchSnapshot();
});

/*Test case for API*/
it("Pizza API test case", async function () {

  global.fetch = jest.fn().mockImplementation(() => {

    var promise = new Promise((resolve, reject) =>{
      resolve({
        json: function () {
          return { pizzas: ["sausage, macroni"] };
        }
      })
    })

    return promise;

  })

  const response = await Pizza.fetchPizzaList();

  expect(response.pizzas).toEqual( ["sausage, macroni"]);

});



/*const fetch = jest.fn(() => {


});

it('render pizza component', () => {
  const PizzaCompo = renderer.create(<PizzaComponent />).toJSON();
  expect(PizzaCompo).toMatchSnapshot();
});

it('renders loading message', () => {
  const { getByText } = render(<PizzaComponent />);
  expect(getByText('Loading data...')).toBeInTheDocument();
});

it('render pizza component', () => {
    const PizzaCompo = renderer.create(<PizzaComponent />).toJSON();
    expect(PizzaCompo).toMatchSnapshot();
});

it('check input field on change', () => {
    const filterIp = renderer.create(<input/>).toJSON();
    expect(filterIp).toMatchSnapshot();
});

test('sortPizzas', () => {


});*/
