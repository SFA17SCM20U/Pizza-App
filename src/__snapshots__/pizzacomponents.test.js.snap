// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading message 1`] = `
<p>
  Loading data...
</p>
`;
